shells/dev-shell/ui-composer.js

```diff
@@ -0,0 +1,9 @@
+import {SlotObserver} from '../lib/xen-renderer.js';
+import {UiSlotComposer} from '../../build/runtime/ui-slot-composer.js';
+
+export const createUiComposer = domRoot => {
+  const composer = new UiSlotComposer();
+  const observer = new SlotObserver(domRoot);
+  composer.observeSlots(observer);
+  return composer;
+};
```

Comment:
Seems unnecessary to pull this into its own file... it's roughly the same amount of code as the bit that has been removed from index.js. I think it's a useful reference for index.js to have a fairly complete, self-contained function that shows how an arc environment is set up and executed.

Reply:
Fair. The middleware I described usually does all that for you, so my aim is to reduce the boilerplate in the `index.js` and leave behind only the novel bits that the particular app wants to customize. 

This is the notion I used in all the other shells, and it's what I would suggest as the best reference example.

Reply:
That may be so, but you're not really removing the boilerplate from DevShell, just making it less visible - which I would argue is less useful here (for reference purposes). If someone wants to write another shell variant, they'll need to do at least all of these steps to get an arc running in a browser.

Reply:
I'm unrolling it for this PR, following your logic.

I'm only saying the actual "least number" of steps is a lot less goal for all the other shells, and my larger goal is to have dev-shell demonstrate the same ergonomics.